belt-dwp/src/ghash.rs

```diff
@@ -0,0 +1,77 @@
+use aead::{KeyInit, KeySizeUser, consts::U1, consts::U16};
+use belt_block::cipher::{BlockSizeUser, ParBlocksSizeUser};
+use universal_hash::{Reset, UhfBackend, UhfClosure, UniversalHash};
+
+use crate::gf::gf128_soft64::Element;
+
+/// GHASH keys (16-bytes)
+pub type Key = universal_hash::Key<GHash>;
+
+/// GHASH blocks (16-bytes)
+pub type Block = universal_hash::Block<GHash>;
+
+/// GHASH tags (16-bytes)
+pub type Tag = universal_hash::Block<GHash>;
+
+#[derive(Clone)]
+pub struct GHash {
```

Comment:
`GHash` is not exposed as part of public API, so you could use inherent methods here without relying on the `universal-hash` dependency.

Reply:
Done, thank you!

Reply:
You only changed the type aliases. I meant that you don't need to implement the `universal-hash` traits and you can remove it from the dependency list.

Reply:
On the second thought, we can leave it as-is with the goal of future migration to the `ghash` crate.